core/src/main/scala/org/apache/spark/scheduler/MapStatus.scala

```diff
@@ -176,15 +179,17 @@ private[spark] object HighlyCompressedMapStatus {
     // From a compression standpoint, it shouldn't matter whether we track empty or non-empty
```

Comment:
Yes, {{RoaringBitmap}} is not symmetric with respect to flips, but that is true also of {{BitSet}}. In Java's BitSet, if you do a flip, can change the {{wordsInUse}} and if you do call trimToSize() afterward, the memory usage would change. What {{runOptimize}} will buy you however is that if you have a small memory usage before the flip (compared to a BitSet), your memory usage will still be "small".